### PR TITLE
make pic inherited for dynamic libraries

### PR DESCRIPTION
**Pull request by John-Colvin**
I've no idea whether this is the "right" approach, but it does seem to work for me.

### CODE REVIEW

**Comment by John-Colvin** [comment]
hmm. So travis doesn't ship ldc with shared libraries...

**Comment by John-Colvin** [comment]
I've disabled the dynamic library tests for ldc, until ldc ships with dynamic druntime as standard.

**Comment by John-Colvin** [comment]
So now this is passing, the question is: is this an acceptable solution to the problem?

**Comment by John-Colvin** [comment]
This has a 1.2.0 milestone but isn't in the beta. Is there an opportunity for it to get in to 1.2.0?

File: test/run-unittest.sh [hunk 1]
```diff
@@ -28,6 +28,12 @@ if [ -z ${DC} ]; then
     DC=dmd
 fi
 
+if [[ $DC -eq "ldc2" ]] || [[ $DC -eq "ldmd2" ]]; then
```
Comment (MartinNowak) [inline test/run-unittest.sh]: DC means ldc2, gdc, or dmd
DMD means ldmd2, gdmd, or dmd

**Comment by MartinNowak** [comment]
Eventually I'd want most libs to just use `library` as target type and have a separate switch/configuration on dub whether to build dynamic or static libraries.
For the explicit dynamic/staticLibrary targets your approach seems to make sense.

**Comment by MartinNowak** [comment]
Cleaned up your run-unittest.sh changes a bit.

**Comment by John-Colvin** [comment]
ping @MartinNowak fixed the conflicts, but of course the auto-merge has been disabled

**Comment by s-ludwig** [comment]
@John-Colvin: as the history is pretty old, can you squash the commits for this PR? Otherwise still looks good to merge.

**Comment by John-Colvin** [comment]
How's this? Did a rebase and some squashing

**Comment by s-ludwig** [comment]
Looks good, thanks!

**Comment by John-Colvin** [comment]
sorry, there was a typo, hopefully it'll work this time...

**Comment by John-Colvin** [comment]
hmm, no, seems like there was a bug in Martin's simplifications, looking in to it now

**Comment by John-Colvin** [comment]
disabled the shared library tests for `gdc` as well

**Comment by John-Colvin** [comment]
good to go now? Only changes since the auto-merge was enabled was deleting a stray `}` in `run-unittest.sh` and adding some `no_build_gdc`s plus a rebase to fix some conflicts

**Comment by s-ludwig** [comment]
Alright, I'll finally press the trigger!

**Comment by John-Colvin** [comment]
Hooray :)

**Comment by s-ludwig** [comment]
I promise I'll have a look at #1002 during the next round!

**Pull request merged**